### PR TITLE
Fix up `InlineArray` tests

### LINKED ISSUE #88861: InlineArrayInvalid needs to be converted to il
In https://github.com/dotnet/runtime/pull/88188 roslyn started emitting errors for the invalid definitions and the test build was disabled.  The test should be coverted to il and reenabled.

### LINKED ISSUE #88957: `Loader/classloader/InlineArray/InlineArrayValid.cs` need to be updated to work with latest roslyn
## Build Information
Build: https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_build/results?buildId=339983
Build error leg or test failing: Build / windows-x64 Checked NativeAOT / Build Tests
Pull request: https://github.com/dotnet/runtime/pull/88188

## Error Message

Fill the error message using [step by step known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssues.md#how-to-fill-out-a-known-issue-error-section).

```json
{
  "ErrorMessage": "",
  "ErrorPattern": "",
  "BuildRetry": false,
  "ExcludeConsoleLog": false
}
```

```
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(184,28): error CS1061: '(object, short)' does not contain a definition for 'o' and no accessible extension method 'o' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(185,28): error CS1061: '(object, short)' does not contain a definition for 's' and no accessible extension method 's' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(201,20): error CS1061: '(object, short)' does not contain a definition for 'o' and no accessible extension method 'o' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(202,20): error CS1061: '(object, short)' does not contain a definition for 's' and no accessible extension method 's' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(209,36): error CS1061: '(object, short)' does not contain a definition for 'o' and no accessible extension method 'o' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(210,40): error CS1061: '(object, short)' does not contain a definition for 's' and no accessible extension method 's' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(232,40): error CS1061: '(object, short)' does not contain a definition for 'o' and no accessible extension method 'o' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(233,44): error CS1061: '(object, short)' does not contain a definition for 's' and no accessible extension method 's' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(245,20): error CS1061: '(object, short)' does not contain a definition for 'o' and no accessible extension method 'o' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(246,20): error CS1061: '(object, short)' does not contain a definition for 's' and no accessible extension method 's' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(253,36): error CS1061: '(object, short)' does not contain a definition for 'o' and no accessible extension method 'o' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(254,40): error CS1061: '(object, short)' does not contain a definition for 's' and no accessible extension method 's' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(259,20): error CS1061: '(object, short)' does not contain a definition for 'o' and no accessible extension method 'o' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(260,20): error CS1061: '(object, short)' does not contain a definition for 's' and no accessible extension method 's' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(294,29): error CS0306: The type 'Span<object>' may not be used as a type argument [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(294,29): error CS0193: The * or -> operator must be applied to a pointer [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(295,31): error CS0306: The type 'Span<object>' may not be used as a type argument [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(295,30): error CS0193: The * or -> operator must be applied to a pointer [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(309,14): error CS0306: The type 'Span<object>' may not be used as a type argument [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(309,13): error CS0193: The * or -> operator must be applied to a pointer [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(429,43): error CS1061: '(object, short)' does not contain a definition for 'o' and no accessible extension method 'o' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(430,47): error CS1061: '(object, short)' does not contain a definition for 's' and no accessible extension method 's' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
```

and @AlekseyTs's [response](https://github.com/dotnet/runtime/pull/88188#issuecomment-1636636617]):
> I looked at several errors, all of them are expected. Compiler no longer uses user-defined indexers for indexing into inline array types.

> For `"error CS1061: '(object, short)' does not contain a definition for 'o' ..."`. Definition of the element field doesn't include tuple element names, but that is the element type that language is using. To fix, add element names to the declaration of the element field. I think custom indexer can be safely deleted.

> For `"error CS0306: The type 'Span' may not be used as a type argument"`. Language doesn't support accessing elements with types that cannot be elements of a span type themselves. A possible workaround is to change indexer to a method and use that to access elements instead of using element access expression syntax for such inline arrays.

#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|0|0|

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
Fixes https://github.com/dotnet/runtime/issues/88957
Fixes https://github.com/dotnet/runtime/issues/88861

/cc @VSadov @mangod9

### CODE REVIEW

**Review by mangod9 — APPROVED** [review]
Thanks for fixing these!

**Comment by VSadov** [comment]
Thanks! I really appreciate help with these. 
Just let me know next time when poaching assigned bugs. I was about to start on these and we could have overlapped. 

That said, getting these out of the way is completely welcome!

**Review by VSadov — APPROVED** [review]
LGTM! Thanks!

File: src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs [hunk 1]
```diff
@@ -109,7 +106,7 @@ struct Arr1
         public E e;
 
         [UnscopedRef]
-        public ref E this[int i] => ref Unsafe.Add(ref e, i);
+        public ref E At(int i) => ref Unsafe.Add(ref e, i);
```
Comment (lambdageek) [inline src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs:109]: nit: `Arr1` doesn't need this `At` function at all. All the tests (as far as i could tell) use indexing - which roslyn generates.

File: src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs [hunk 2]
```diff
@@ -171,7 +168,7 @@ struct ObjShortArr
         public (object, short) element;
 
         [UnscopedRef]
-        public ref (object o, short s) this[int i] => ref Unsafe.Add(ref element, i);
+        public ref (object o, short s) At(int i) => ref Unsafe.Add(ref element, i);
```
Comment (lambdageek) [inline src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs:171]: I don't think this is necessary.  From #86183 

 >Definition of the element field doesn't include tuple element names, but that is the element type that language is using. To fix, add element names to the declaration of the element field. I think custom indexer can be safely deleted.

So something like this:

```csharp
public (object o, short s) element;
```

and remove `At` and replace `arr.At(i).o` by `arr[i].o` (and same for `.s`) in the uses, below

Comment (AaronRobinsonMSFT) [reply src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs:171]: This doesn't seem to work.

```
.../runtime/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(178,28): error CS1061: '(object, short)' does not contain a definition for 'o' and no accessible extension method 'o' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [.../runtime/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj]
.../runtime/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(179,28): error CS1061: '(object, short)' does not contain a definition for 's' and no accessible extension method 's' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [.../runtime/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj]
```

```csharp
    [InlineArray(Length)]
    struct ObjShortArr
    {
        public const int Length = 100;
        public (object, short) element;

        // [UnscopedRef]
        // public ref (object o, short s) At(int i) => ref Unsafe.Add(ref element, i);

        [MethodImpl(MethodImplOptions.NoInlining)]
        public static ObjShortArr CreateArray(int recCount) {
            if (recCount > 0) {
                return CreateArray(recCount-1);
            } else {
                var arr = new ObjShortArr();
                for (short i = 0; i < ObjShortArr.Length; i++)
                {
                    arr[i].o = i;
                    arr[i].s = (short)(i + 1);
                }
                return arr;
            }
        }
    }
```

Comment (AaronRobinsonMSFT) [reply src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs:171]: I missed the `o` and `s`.

**Comment by AaronRobinsonMSFT** [comment]
This is a container timeout and unrelated.

**Pull request merged**